### PR TITLE
Added missing kind parameter required when adding a partition key

### LINKED ISSUE #3991: az cosmosdb collection create fails when specifying a custom partition
### Description
Using the command:
```
az cosmosdb collection create --name --db-name --resource-group --collection-name --key --partition-key-path "/StoreUid"
```
This command results in an error including:
```
DEBUG: Status code: 400
{"code":"BadRequest","message":"Message: {"Errors":["The 'kind' value '' specified in the partition key definition is invalid. Please choose between 'Hash' and 'Range' partition
types."]}\r\

ERROR: Operation Failed: Invalid Arg {"Errors":["The 'kind' value '' specified in the partition key definition is invalid. Please choose between 'Hash' and 'Range' partition types."]}
```
Upon investigation I noticed that the request is malformed and missing the the kind value in command_modules, azure-cli-cosmosdb, azure, cli, command_modules, cosmosdb, custom. in def _populate_collection_definition

---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: MSI

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here: 2.0.9

**OS Version:** What OS and version are you using?  
Answer here: Windows 10 Creators Update

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: PowerShell

### PR DESCRIPTION
**Pull request by CameronVetter**
Fixes #3991 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by troydai** [comment]
It would be nice if the change comes with a test.

**Comment by CameronVetter** [comment]
Agreed that there was a missing test which is what allowed this bug through to begin with.  There is quite a bit of friction for me to be able to write the test since I'm not a python developer, otherwise I would have done it right away.  Is this going to be blocked until I write a test?

**Comment by troydai** [comment]
It's fine. I'd like to have @moderakh take a look your PR. Once @moderakh approves this PR it's good to go.

**Comment by moderakh** [comment]
@CameronVetter 
if the partition key kind is not explicitly specified the default of 'Hash' will be used.

Any particular reason that you explicitly specified the partition key kind in the code? Were you getting failure without this?

@troydai thanks for tagging me.

**Comment by CameronVetter** [comment]
Take a look at issue #3991 I have a cut and paste of the error message with v 2.0.9, it appears that the REST interface is defaulting to an empty string without the property explicitly specified in JSON of the call.

**Comment by moderakh** [comment]
It seems there is recent change in behavour in the cosmosdb endpoint. And due to that partition key kind must be present in the collection definition.

@CameronVetter thank you for reporting this, and thanks for the fix.

File: src/command_modules/azure-cli-cosmosdb/HISTORY.rst [hunk 1]
```diff
@@ -1,5 +1,10 @@
 .. :changelog:
 
+unreleased
+===============
+
+* Minor fix allowing Creation of Collection with custom partition key
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-cosmosdb/HISTORY.rst]: This should be under the `Release History` heading and should use `++++` instead of `=====`

Comment (CameronVetter) [inline src/command_modules/azure-cli-cosmosdb/HISTORY.rst]: I have made the requested changes, but I want to point out I followed the guidelines here:  https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log

Those probably should be updated to represent the new standards or modified to clarify what was intended. I'm still not 100% sure that I did what was intended, tried to read between the lines a bit.

Comment (CameronVetter) [inline src/command_modules/azure-cli-cosmosdb/HISTORY.rst]: @derekbekoe could you review this again please?

Comment (troydai) [inline src/command_modules/azure-cli-cosmosdb/HISTORY.rst]: Thanks for point this out. We will update the doc. I'll approve and merge this change.

**Comment by SamVanhoutte** [comment]
Hello all, I need exactly this fix for my vsts release management.  Do you have any idea when this will be rolled out onto the VSTS servers?  Is there a way to workaround this issue, until then?  (exception when I want to create a collection with a partition property)

**Comment by CameronVetter** [comment]
I did not find any workaround for this, my scripts currently fail and we have to manually do these steps.  I'm also very eager to see this roll into the CLI...

**Pull request merged**